public/components/integrations/components/available_integration_overview_page.tsx

```diff
@@ -0,0 +1,250 @@
+  showModal: (input: string) => void;
+  isCardView: boolean;
+  setCardView: (input: boolean) => void;
+  renderCateogryFilters: () => React.JSX.Element;
+}
+
+export interface AvailableIntegrationsList {
+  hits: AvailableIntegrationType[];
+}
+
+export interface AvailableIntegrationsCardViewProps {
+  data: AvailableIntegrationsList;
+  showModal: (input: string) => void;
+  isCardView: boolean;
+  setCardView: (input: boolean) => void;
+  query: string;
+  setQuery: (input: string) => void;
+  renderCateogryFilters: () => React.JSX.Element;
+}
+
+export function AvailableIntegrationOverviewPage(props: AvailableIntegrationOverviewPageProps) {
+  const { chrome, http } = props;
+
+  const [query, setQuery] = useState('');
+  const [isCardView, setCardView] = useState(true);
+  const { setToast } = useToast();
+  const [data, setData] = useState<AvailableIntegrationsList>({ hits: [] });
+
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [modalLayout, setModalLayout] = useState(<EuiOverlayMask />);
+
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+
+  const onButtonClick = () => {
+    setIsPopoverOpen(!isPopoverOpen);
+  };
+
+  const closePopover = () => {
+    setIsPopoverOpen(false);
+  };
+
+  const [items, setItems] = useState([
+    { name: 'http' },
+    { name: 'logs' },
+    { name: 'communication' },
+    { name: 'cloud' },
+    { name: 'aws_elb' },
+    { name: 'container' },
+  ]);
+
+  function updateItem(index) {
+    if (!items[index]) {
+      return;
+    }
+
+    const newItems = [...items];
+
+    switch (newItems[index].checked) {
+      case 'on':
+        newItems[index].checked = undefined;
+        break;
+
+      default:
+        newItems[index].checked = 'on';
+    }
+
+    setItems(newItems);
+  }
+
+  const helper = items.filter((item) => item.checked === 'on').map((x) => x.name);
+
+  const button = (
+    <EuiFilterButton
+      iconType="arrowDown"
+      onClick={onButtonClick}
+      isSelected={isPopoverOpen}
+      numFilters={items.length}
+      hasActiveFilters={!!items.find((item) => item.checked === 'on')}
+      numActiveFilters={items.filter((item) => item.checked === 'on').length}
+    >
+      Categories
+    </EuiFilterButton>
+  );
+
+  const getModal = (name: string) => {
+    setModalLayout(
+      getAddIntegrationModal(
+        () => {
+          addIntegrationRequest(name);
+          setIsModalVisible(false);
+        },
+        () => {
+          setIsModalVisible(false);
+        },
+        'Name',
+        'Namespace',
+        'Tags (optional)',
+        name,
+        'prod',
+        'Add Integration Options',
+        'Cancel',
+        'Add',
+        'test'
+      )
+    );
+    setIsModalVisible(true);
+  };
+
+  useEffect(() => {
+    chrome.setBreadcrumbs([
+      {
+        text: 'Integrations',
+        href: '#/',
+      },
+    ]);
+    handleDataRequest();
+  }, []);
+
+  async function handleDataRequest() {
```

Comment:
Minor: Any particular reason we keep two styles of declaring functions? Ex.
`const getModal = (name: string)` is declared as a function object.
`async function handleDataRequest()` is declared as a JS function
Keeping same style is good for readability.

Reply:
Good point - it might be because two people were developing 😂 , but yes this is a good suggestion - I created a follow up issue here: https://github.com/opensearch-project/dashboards-observability/issues/623#issuecomment-1631107114.